src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -264,19 +275,36 @@ internal RegionVersionOne CreateRegion(Region v2Region)
             return region;
         }
 
-        internal ReplacementVersionOne CreateReplacement(Replacement v2Replacement)
+        internal ReplacementVersionOne CreateReplacement(Replacement v2Replacement, Uri uri)
         {
             ReplacementVersionOne replacement = null;
 
             if (v2Replacement != null)
             {
                 replacement = new ReplacementVersionOne();
+                FileContent insertedContent = v2Replacement.InsertedContent;
 
-                if (v2Replacement.InsertedContent != null)
+                if (insertedContent != null)
                 {
-                    replacement.InsertedBytes = v2Replacement.InsertedContent.Text != null ?
-                                                    SarifUtilities.GetUtf8Base64String(v2Replacement.InsertedContent.Text) :
-                                                    v2Replacement.InsertedContent.Binary;
+                    if (insertedContent.Text != null)
+                    {
+                        FileData fileData = _currentV2Run.Files[uri.OriginalString];
```

Comment:
There might not be an entry for this file in the `run.files` dictionary, and if there is, it might not have an `encoding` property. The logic needs to be:
```
IF the run.files dictionary contains an entry for this file
    AND the entry contains an encoding property
    AND the encoding is known to .NET
THEN
    Compute the inserted bytes
ELSE
    throw
END
```

Reply:
And given that you might not know the encoding at all, I propose `UnknownEncodingException` instead of `UnsupportedEncodingException`. Your call.

Reply:
Really, this calls for four unit tests, one for each case:
1. There is no dictionary entry.
2. There is a dictionary entry but it has no encoding.
3. It has an encoding but .NET doesn't recognize it.
4. It has a recognized encoding.

You test currently covers #&#xFEFF3; and #&#xFEFF4;.

Reply:
Done